driver-sync/src/test/functional/com/mongodb/client/CrudProseTest.java

```diff
@@ -16,181 +16,411 @@
-            // effective
-            assertTrue(e.getMessage().contains("Write error"));
-            assertNotNull(e.getError().getDetails());
-            if (serverVersionAtLeast(5, 0)) {
-                assertFalse(e.getError().getDetails().isEmpty());
+    void testWriteErrorDetailsIsPropagated() {
+        try (MongoClient client = createMongoClient(getMongoClientSettingsBuilder())) {
+            MongoCollection<Document> collection = droppedCollection(client, Document.class);
+            droppedDatabase(client).createCollection(
+                    collection.getNamespace().getCollectionName(),
+                    new CreateCollectionOptions().validationOptions(new ValidationOptions().validator(Filters.type("x", "string"))));
+            assertAll(
+                    () -> {
+                        MongoWriteException actual = assertThrows(MongoWriteException.class, () ->
+                                collection.insertOne(new Document("x", 1)));
+                        // These assertions don't do exactly what's required by the specification,
+                        // but it's simpler to implement and nearly as effective.
+                        assertTrue(actual.getMessage().contains("Write error"));
+                        assertNotNull(actual.getError().getDetails());
+                        if (serverVersionAtLeast(5, 0)) {
+                            assertFalse(actual.getError().getDetails().isEmpty());
+                        }
+                    },
+                    () -> {
+                        MongoBulkWriteException actual = assertThrows(MongoBulkWriteException.class, () ->
+                                collection.insertMany(singletonList(new Document("x", 1))));
+                        // These assertions don't do exactly what's required by the specification,
+                        // but it's simpler to implement and nearly as effective.
+                        assertTrue(actual.getMessage().contains("Write errors"));
+                        assertEquals(1, actual.getWriteErrors().size());
+                        if (serverVersionAtLeast(5, 0)) {
+                            assertFalse(actual.getWriteErrors().get(0).getDetails().isEmpty());
+                        }
+                    }
+            );
+
+        }
+    }
+
+    @DisplayName("3. MongoClient.bulkWrite batch splits a writeModels input with greater than maxWriteBatchSize operations")
+    @Test
+    void testBulkWriteSplitsWhenExceedingMaxWriteBatchSize() {
+        assumeTrue(serverVersionAtLeast(8, 0));
+        assumeFalse(isServerlessTest());
+        assumeTrue(Runtime.getRuntime().availableProcessors() < 1, "BULK-TODO implement prose test https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.md#3-mongoclientbulkwrite-batch-splits-a-writemodels-input-with-greater-than-maxwritebatchsize-operations");
```

Comment:
Same as previous - in CSOT we'd add a jira ticket to track the todos there.

Reply:
The reply is also the same:) https://github.com/mongodb/mongo-java-driver/pull/1486#discussion_r1752076281